PowerShell/ScubaGear/Testing/Unit/PowerShell/Providers/AADProvider/LoadObjectDataIntoPrivilegedUserHashtable.Tests.ps1

```diff
@@ -0,0 +1,191 @@
+            $ObjectId = [Guid]::NewGuid().Guid  # Random GUID for ObjectId
+            $TenantHasPremiumLicense = $true
+            $M365Environment = "commercial"
+
+            # Mock Get-MgBetaDirectoryObject to return a user-type object
+            Mock Get-MgBetaDirectoryObject {
+                [PSCustomObject]@{
+                    AdditionalProperties = @{
+                        "@odata.type" = "#microsoft.graph.user"  # Simulates a user type
+                    }
+                }
+            }
+
+            # Mock Get-MgBetaUser to return a user with DisplayName and OnPremisesImmutableId
+            Mock Get-MgBetaUser {
+                [PSCustomObject]@{
+                    DisplayName            = "John Doe"
+                    OnPremisesImmutableId  = "ABC123"
+                }
+            }
+
+            # Test 1 - Do NOT pass ObjectType
+            LoadObjectDataIntoPrivilegedUserHashtable -RoleName $RoleName -PrivilegedUsers $PrivilegedUsers -ObjectId $ObjectId -TenantHasPremiumLicense $TenantHasPremiumLicense -M365Environment $M365Environment
+            # Assertions to ensure the user was processed correctly
+             $PrivilegedUsers[$ObjectId].DisplayName | Should -Be "John Doe"
+             $PrivilegedUsers[$ObjectId].OnPremisesImmutableId | Should -Be "ABC123"
+             $PrivilegedUsers[$ObjectId].roles | Should -Contain $RoleName
+
+            # Test 2 - Pass ObjectType
+            $PrivilegedUsers = @{}
+            $Objecttype = "user"
+            LoadObjectDataIntoPrivilegedUserHashtable -Objecttype $Objecttype -RoleName $RoleName -PrivilegedUsers $PrivilegedUsers -ObjectId $ObjectId -TenantHasPremiumLicense $TenantHasPremiumLicense  -M365Environment $M365Environment
+
+            # Assertions to ensure the user was processed correctly
+            $PrivilegedUsers[$ObjectId].DisplayName | Should -Be "John Doe"
+            $PrivilegedUsers[$ObjectId].OnPremisesImmutableId | Should -Be "ABC123"
+            $PrivilegedUsers[$ObjectId].roles | Should -Contain $RoleName
+        }
+
+
+        It 'Objecttype is a group' {
+            # Set up the parameters for the test
+            $RoleName = "Global Administrator"  # Mock role
+            $PrivilegedUsers = @{}  # Empty hashtable for privileged users
+            $ObjectId = [Guid]::NewGuid().Guid  # Random GUID for ObjectId, simulating a group ID
+            $TenantHasPremiumLicense = $true
+            $M365Environment = "commercial"
+
+            # Mock Get-MgBetaDirectoryObject to return a group-type object
+            Mock Get-MgBetaDirectoryObject {
+                [PSCustomObject]@{
+                    AdditionalProperties = @{
+                        "@odata.type" = "#microsoft.graph.group"  # Simulates a group type
+                    }
+                }
+            }
+
+            # Mock Get-MgBetaGroupMember to return two group members (users)
+            Mock Get-MgBetaGroupMember {
```

Comment:
See my other comment on use of Get-MgBetaGroupMember. During my testing I didn't get incorrect results when assigning a new user to a group as an "owner" role type, but perhaps we mock Get-MgBetaGroupOwner as well for complete test coverage.

Reply:
Discussed offline, resolving conversation